### DISCUSSION TITLE
Is there any SDK for developing VSCode Extensions?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#345] [Extension Development QnA] [answered]
Is there any SDK for developing VSCode Extensions?

**Comment by alefragnani** [comment, marked as answer]
Hi @liaodalin19903 ,

Yes, there is. 

You can start looking here https://code.visualstudio.com/api, for all things about extensibility, where you will find not only [Guides](https://code.visualstudio.com/api/extension-guides/overview), but [Samples](https://github.com/microsoft/vscode-extension-samples), [Capabilities](https://code.visualstudio.com/api#what-can-extensions-do), [How to Build](https://code.visualstudio.com/api#how-to-build-extensions) and a bunch of additional resources. 

If you looking exclusively for APIs, you could look [APIs](https://code.visualstudio.com/api/references/vscode-api) and [Contribution Points](https://code.visualstudio.com/api/references/contribution-points).

Which such documentation and references, you will have everything you need to start your first extension.

Hope this helps
